lib/wallets/WalletService.ts

```diff
@@ -0,0 +1,47 @@
+import { Cell } from "ton";
+import { TonhubConnector } from "ton-x";
+import { TransactionDetails } from "../transaction-sender";
+
+import { TonhubWalletAdapter } from "./adapters/TonhubWalletAdapter";
+import { TonWalletWalletAdapter } from "./adapters/TonWalletAdapter";
+import { WalletAdapter, TransactionRequest, Wallet, Adapters } from "./types";
+
+const IS_TESTNET = false;
+
+export class WalletService {
+  private readonly adapters: Map<string, WalletAdapter<any>> = new Map();
+
+  registerAdapter(adapterId: string, adapter: WalletAdapter<any>) {
+    console.log(this.adapters);
+
+    this.adapters.set(adapterId, adapter);
+  }
+
+  createSession<S>(adapterId: string, appName: string): Promise<S> {
+    const adapter = this.adapters.get(adapterId) as WalletAdapter<S>;
+    return adapter.createSession(appName);
+  }
+
+  async awaitReadiness<S>(adapterId: string, session: S): Promise<Wallet> {
+    const adapter = this.adapters.get(adapterId) as WalletAdapter<S>;
+    return adapter.awaitReadiness(session);
+  }
+
+  async getWallet<S>(adapterId: string, session: S): Promise<Wallet> {
+    const adapter = this.adapters.get(adapterId) as WalletAdapter<S>;
+    return adapter.getWallet(session);
+  }
+
+  async requestTransaction(adapterId: string, session: any, request: TransactionDetails, onSuccess?: () => void): Promise<void | boolean> {
+    const adapter = this.adapters.get(adapterId) as WalletAdapter<S>;
+
+    return adapter.requestTransaction(session, request, onSuccess);
+  }
+}
+
+export const walletService = new WalletService();
+
+const tonhubConnector = new TonhubConnector({ testnet: IS_TESTNET });
```

Comment:
1. isn't it strange that the generic class file contains the specific details for instantiating the tonhub connector? do you think we should move it to a tonhub file?
2. will this work in sandbox? (A - testnet is always false, B - afaik tonhub now works mostly with sandbox and not testnet)